app/main.rs

```diff
@@ -996,6 +996,7 @@ async fn main() -> Result<()> {
     let validator = Validator::new(
         mainchain_client.clone(),
         mainchain_rest_client,
+        cli.clone().node_blocks_dir_opts.dir,
```

Comment:
You don't need to clone `cli` here, could just do `cli.node_blocks_dir_opts.dir.clone()`,